lib/core/lib/ros/url_builder.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Ros
+  module UrlBuilder
+    class JSONAPI::LinkBuilder
+
+      def base_url
+        return @base_url if service_engine_name.blank?
+
+        "#{@base_url}/#{service_engine_name}"
+      end
+
+      private
+
+      def service_engine_name
+        primary_resource_klass.superclass.to_s.split("::")[0...-1].first&.downcase
```

Comment:
I'm not sure if this could be an issue here, but should we use `.underscore`?
Usage of `.downcase` will give us `SomeName` => `somename`, but usage of `.underscore` will give `SomeName` => `some_name`.

Reply:
What you say seems to make sense. If we go ahead with this monkey patching, then we'll need to handle that use case.

Reply:
agree my bad